### ISSUE TITLE
[question] Conan search only shows packages that are outdated from recipe

### ISSUE THREAD

**Issue by fjp** [conan-io/conan#14010]
### What is your question?

Hi @memsharded I could observe with conan 1.45 and also 1.60 that when I am searching for a reference on a remote artifactory I only see outdated packages even though on the artifactory there is a new recipe revision which contains one non outdated package.

Th search command that reveals only outdated packages is the following:

````
$ conan search mypackage/1.0.1@user/stable -r remote-dev
Existing packages for recipe mypackage/1.0.1@user/stable:

Existing recipe in remote 'remote-dev':

    Package_ID: 0c7ac9a529c76b3697dd6c8b08eaea9cbbd48e44
        [options]
            shared: False
        [settings]
            arch: armv7hf
            build_type: Debug
            compiler: gcc
            compiler.libcxx: libstdc++11
            compiler.version: 9.3
            os: Linux
        Outdated from recipe: True

    Package_ID: 1299b26cefc5e04411a3481abe1c8b63227e2896
        [options]
            shared: False
        [settings]
            arch: x86
            build_type: Release
            compiler: Visual Studio
            compiler.runtime: MD
            compiler.version: 14
            os: Windows
        Outdated from recipe: True

    Package_ID: 14572928e5faf7a330814917e884bbf87f1d8b5d
        [options]
            shared: False
        [settings]
            arch: armv7hf
            build_type: Release
            compiler: gcc
            compiler.libcxx: libstdc++11
            compiler.version: 9.3
            os: Linux
        Outdated from recipe: True

    Package_ID: 251974cfdccce6efd38e10feccec3c61142d3a33
        [options]
            shared: False
        [settings]
            arch: armv7hf
            build_type: Release
            compiler: gcc
            compiler.libcxx: libstdc++
            compiler.version: 4.9
            os: Linux
        Outdated from recipe: True
````

Using the `--revision` flag shows both old and new recipe revisions on artifactory:

```
$ conan searchmypackage/1.0.1@user/stable -r remote-dev --revision
Revisions for 'mypackage/1.0.1@user/stable' at remote 'remote-dev':
eb650b8e69461657d08a298e6f4f81f0 (2023-05-31 00:09:44 UTC)
52c9fc7da5009c0e11d4cddf9bff688e (2022-09-08 10:39:34 UTC)
```

I thought conan search would by default return the packages from the latest recipe revision (eb650b8e69461657d08a298e6f4f81f0), if no revision is specified explicitly, and not just show the ones from an older recipe revision (52c9fc7da5009c0e11d4cddf9bff688e)? The `--outdated` flag seems to make no difference. With or without this flag I get only all outdated recipes from the old revision.

Only when I specify the latest recipe revision I get the search result I would have expected in the first place:

```
$ conan search mypackage/1.0.1@user/stable#eb650b8e69461657d08a298e6f4f81f0 -r remote-dev
Existing packages for recipe mypackage/1.0.1@user/stable:

Existing recipe in remote 'remote-dev':

    Package_ID: 2148b953e915707b8702b3a3c8eac94b4a9f6f37
        [options]
            shared: False
        [settings]
            arch: armv8
            build_type: Release
            compiler: gcc
            compiler.libcxx: libstdc++11
            compiler.version: 9
            os: Linux
        Outdated from recipe: False
```

```
$ conan search mypackage/1.0.1@user/stable#eb650b8e69461657d08a298e6f4f81f0 -r remote-dev --outdated
Existing packages for recipe mypackage/1.0.1@user/stable:

Existing recipe in remote 'remote-dev':

There are no outdated packages for reference 'mypackage/1.0.1@user/stable', but package recipe found.
```

### Have you read the CONTRIBUTING guide?

- [X] I've read the CONTRIBUTING guide

**Comment by memsharded** [comment]
Hi @fjp 

Thanks for your question.

The truth is that the server side implementation was not intended to simultaneously keep serving packages for clients with revisions and without revisions for a long time in production.
The logic is that if a client didn't activate the revisions, is that they do not care about the revisions, and they are fine with whatever binary is provided to them. In practice is like collapsing all the binaries of all revisions into a single "0" revisions. Doing it that way allowed users that were used to continuously overwrite binaries without revisions to transition to use revisions with less pain (not needing to suddenly have to re-build all the binaries for every new recipe-revision). 

But it is true that this behavior, while it was good to help the transition, is not the best if you try to run in production for longer times with clients without revisions. The very strong recommendation is to enable revisions everywhere and not look back... :)

**Comment by fjp** [comment]
Thanks @memsharded for the insight. However as far as I know my conan client has revisions enabled, at least it seems so in my conan config:

```
[log]
run_to_output = True        # environment CONAN_LOG_RUN_TO_OUTPUT
run_to_file = False         # environment CONAN_LOG_RUN_TO_FILE
level = critical            # environment CONAN_LOGGING_LEVEL
print_run_commands = False  # environment CONAN_PRINT_RUN_COMMANDS

[general]
default_profile = default
compression_level = 9                 # environment CONAN_COMPRESSION_LEVEL
sysrequires_sudo = True               # environment CONAN_SYSREQUIRES_SUDO
request_timeout = 60                  # environment CONAN_REQUEST_TIMEOUT (seconds)
default_package_id_mode = semver_direct_mode # environment CONAN_DEFAULT_PACKAGE_ID_MODE
use_always_short_paths = True
revisions_enabled = 1

[storage]
path = ./data

[proxies]

[hooks]
attribute_checker
```

Also the remote has it enabled right? Otherwise it wouldn't have the two folders reciepe revision folders. Or how can I check if the remote and my conan client has revisions really enabled?

> In practice is like collapsing all the binaries of all revisions into a single "0" revisions.

Please note that the package id 2148b953e915707b8702b3a3c8eac94b4a9f6f37 from the new revision eb650b8e69461657d08a298e6f4f81f0 is not part of the plain conan serach (the one without specifying the recipe revision). From your explanation I would have expected all packages listed (outdated and non-outdated ones) but I see only the outdated ones stored under the old recipe revision. Can you please explain why I see this behavior?

**Comment by memsharded** [comment]
Yes, the remote support for v2 is always enabled. What version are you running in the server?

In theory it is impossible to get ``Outdated from recipe: True`` with revisions enabled. I have done a few checks, but couldn't reproduce it, that is unexpected. Could you please re-check?

**Comment by fjp** [comment]
It is version 7.41.14 rev and I can confirm it is still happening for multiple developers. Even after uploading more binary packages to the latest revision. 

```
$ conan search mypackage/1.0.1@user/stable -r remote-dev
```

returns only the binary packages that are outdated (`Outdated from recipe: True`)

![image](https://github.com/conan-io/conan/assets/1286618/a9e09050-72ce-4ab5-a2bb-26bca674ece1)

**Comment by fjp** [comment]
@memsharded any ideas why the search only returns packages where outdated from recipe is True? Could it be related to the server version 7.41.14?

**Comment by memsharded** [comment]
Sorry this was not followed up back then.

Now this ticket seems outdated, as it was referring to Conan 1 "outdated from recipe" prior to revisions, no longer an issue in Conan2. So closing this ticket, but please comment or create new tickets for any further question or issue, thanks for the feedback!
